### PR TITLE
Actions runs cli

### LINKED ISSUE #1479: Hooks: CLI to list and output actions and hooks

### LINKED ISSUE #1502: Validate branch and run id existance if neede
Actions should verify that the requested repository, branch, and run ID exist and not assume empty results in some cases.

### PR DESCRIPTION
**Pull request by nopcoder**
Closes #1479 
Fixes #1502

- lakectl actions runs list lakefs://repo1
- lakectl actions runs list lakefs://repo1@master
- lakectl actions runs show lakefs://repo1 runID1
- lakectl cat-hook-output lakefs://repo1 runID1 hookRunID1 (hidden)

### CODE REVIEW

File: api/controller.go [hunk 1]
```diff
@@ -2804,7 +2815,28 @@ func (c *Controller) ActionsListRunsHandler() actionsop.ListRunsHandler {
...
+		if errors.Is(err, catalog.ErrRepositoryNotFound) {
+			return actionsop.NewListRunsNotFound().
+				WithPayload(responseErrorFrom(catalog.ErrRepositoryNotFound))
+		}
+		if err != nil {
+			return actionsop.NewListRunsDefault(http.StatusInternalServerError).
+				WithPayload(responseErrorFrom(err))
+		}
+		if params.Branch != nil {
+			exists, err := deps.Cataloger.BranchExists(deps.ctx, params.Repository, swag.StringValue(params.Branch))
```
Comment (itaiad200) [inline api/controller.go:2877]: Isn't this check cover the Repo check too?

File: config/config.go [hunk 2]
```diff
@@ -200,38 +197,6 @@ func (c *Config) GetDatabaseParams() dbparams.Database {
 	}
 }
 
-type AwsS3RetentionConfig struct {
```
Comment (itaiad200) [inline config/config.go]: It seems your PR is mixed up with retention deletion commit.

File: cmd/lakectl/cmd/runs_list.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		}
+
+		Write(actionsRunsListTemplate, data)
+	},
+}
+
+//nolint:gochecknoinits
+func init() {
+	actionsRunsCmd.AddCommand(runsListCmd)
+	runsListCmd.Flags().Int("amount", -1, "how many results to return, or-1 for all results (used for pagination)")
```
Comment (itaiad200) [inline cmd/lakectl/cmd/runs_list.go:88]:
```suggestion
	runsListCmd.Flags().Int("amount", -1, "how many results to return, or '-1' for all results (used for pagination)")
```

File: cmd/lakectl/cmd/runs_list.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+					"End Time",
+					"Branch",
+					"Commit ID",
+				},
+				Rows: rows,
+			},
+		}
+		if pagination != nil && swag.BoolValue(pagination.HasMore) {
+			data.Pagination = &Pagination{
+				Amount:  amount,
```
Comment (itaiad200) [inline cmd/lakectl/cmd/runs_list.go:75]: amount isn't returned if this is the last/only frame?

File: cmd/lakectl/cmd/runs_list.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+				row.Status,
+			}
+		}
+
+		data := struct {
+			ActionsRunsTable *Table
+			Pagination       *Pagination
+		}{
+			ActionsRunsTable: &Table{
+				Headers: []interface{}{
```
Comment (itaiad200) [inline cmd/lakectl/cmd/runs_list.go:62]: `Status` is missing

File: cmd/lakectl/cmd/runs_show.go [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	Status: {{ $val.Status }}
+Output:
+{{ printf $val.HookRunID | call $log }}
+{{ end }}{{ .Pagination | paginate }}
+`
+
+const runsShowRequiredArgs = 2
+
+var runsShowCmd = &cobra.Command{
+	Use:     "show",
```
Comment (itaiad200) [inline cmd/lakectl/cmd/runs_show.go:40]: `show` is too general IMO. Maybe `describe`?

File: cmd/lakectl/cmd/runs_show.go [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+		if err != nil {
+			DieErr(err)
+		}
+		data := struct {
+			Hooks      []*models.HookRun
+			HookLog    func(hookRunID string) (string, error)
+			Pagination *Pagination
+		}{
+			Hooks: response,
+			HookLog: func(hookRunID string) (string, error) {
```
Comment (itaiad200) [inline cmd/lakectl/cmd/runs_show.go:75]: I think this has the potential to mess stdout and make the command less accessible. Maybe just log the actual command for fetching the output?
`lakectl cat-hook-output lakefs://u.Repository runID hookRunID `

File: cmd/lakectl/cmd/runs_show.go [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+			DieErr(err)
+		}
+		Write(actionTaskResultTemplate, data)
+	},
+}
+
+//nolint:gochecknoinits
+func init() {
+	actionsRunsCmd.AddCommand(runsShowCmd)
+	runsShowCmd.Flags().Int("amount", -1, "how many results to return, or-1 for all results (used for pagination)")
```
Comment (itaiad200) [inline cmd/lakectl/cmd/runs_show.go:102]:
```suggestion
	runsShowCmd.Flags().Int("amount", -1, "how many results to return, or '-1' for all results (used for pagination)")
```

**Review by itaiad200 — CHANGES_REQUESTED** [review]
Almost there, mostly minor comments.

File: api/controller.go (hunk 1 above)
Comment (nopcoder) [reply api/controller.go:2877]: true, but you will get branch not found in this case

File: config/config.go (hunk 2 above)
Comment (nopcoder) [reply config/config.go]: let me know if you still see it after the merge with master.

File: cmd/lakectl/cmd/runs_list.go (hunk 5 above)
Comment (nopcoder) [reply cmd/lakectl/cmd/runs_list.go:62]: Thanks, missed this one.

File: cmd/lakectl/cmd/runs_show.go (hunk 6 above)
Comment (nopcoder) [reply cmd/lakectl/cmd/runs_show.go:40]: wanted something like the current show command, `show         See detailed information about an entity by ID (commit, user, etc)`.

File: cmd/lakectl/cmd/runs_show.go (hunk 7 above)
Comment (nopcoder) [reply cmd/lakectl/cmd/runs_show.go:75]: Thought about it - this one after discussion with Oz.
Keeping the `cat-hook-output` hidden command and have a simple one command to get all the information.
so the end-user life will be easy

File: cmd/lakectl/cmd/runs_show.go (hunk 8 above)
Comment (nopcoder) [reply cmd/lakectl/cmd/runs_show.go]: Updated all the 'amount' flags. Also changed from all results to default as we do not iterate over everything with -1

File: api/client.go [hunk 9]
```diff
@@ -92,6 +93,11 @@ type RepositoryClient interface {
 
 	RefsDump(ctx context.Context, repository string) (*models.RefsDump, error)
 	RefsRestore(ctx context.Context, repository string, manifest *models.RefsDump) error
+
```
Comment (itaiad200) [inline api/client.go:96]: You can also add ListCommitRuns

Comment (itaiad200) [reply api/client.go:96]: Here and as command I mean

File: cmd/lakectl/cmd/runs_list.go [hunk 10]
```diff
@@ -3,6 +3,8 @@ package cmd
 import (
 	"context"
 
+	"github.com/treeverse/lakefs/api/gen/models"
+
```
Comment (itaiad200) [inline cmd/lakectl/cmd/runs_list.go:7]: /dd

**Review by itaiad200 — APPROVED** [review]
Looks Great!

**Pull request merged**